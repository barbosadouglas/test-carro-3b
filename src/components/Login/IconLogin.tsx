import React from "react";

import {
  StyleSheet,
  View,
  Image,
  Button,
  TouchableOpacity,
} from "react-native";

interface IconLoginProps {
  click: () => void;
}
export default function IconLogin({ click }: IconLoginProps) {
  return (
    <View>
      <TouchableOpacity onPress={click}>
        <Image
          source={require("../../../assets/images/login.png")}
          style={styles.image}
        />
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  image: {
    backgroundColor: "gray",
    alignItems: "center",

    borderRadius: 96,
  },
});
