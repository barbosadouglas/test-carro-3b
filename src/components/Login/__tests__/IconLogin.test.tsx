import React from "react";
import renderer from "react-test-renderer";
import IconLogin from "../IconLogin";
//teste snapshot do jest, para garantir a renderização do componente
describe("IconLogin Component", () => {
  //Verificar se o componente é renderizado corretamente
  it("renders correctly", () => {
    //Criar uma instancia do componente IconLogin com uma função click simulada
    const tree = renderer.create(<IconLogin click={() => {}} />).toJSON();
    //Uso do snapshot para verificar se as renderizações são iguais.
    expect(tree).toMatchSnapshot();
  });
});
