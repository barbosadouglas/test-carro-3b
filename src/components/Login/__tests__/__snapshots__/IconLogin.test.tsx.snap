// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLogin Component renders correctly 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "gray",
          "borderRadius": 96,
        }
      }
    />
  </View>
</View>
`;
